src/proj_types.jl

```diff
@@ -0,0 +1,46 @@
+## Types
+
+# Projection context.  TODO: Will this be exposed?
+#type Context
+#    rep::Ptr{Void} # Pointer to internal projCtx struct
+#end
+
+@doc """
+Cartographic projection type
+""" ->
+type Projection
+    #ctx::Context   # Projection context object
+    rep::Ptr{Void} # Pointer to internal projPJ struct
+    
+    # [geod]: a structure containing the parameters of the spheroid
+    # some of the fields in [geod] are mildly duplicative of the information
+    # available in [rep], which can be exposed only through _get_spheroid_defn
+    geod::Ptr{Void}
```

Comment:
I suggest simply adding a dummy geod_geodesic type when the version of the underlying libproj isn't enough:

```julia
type geod_geodesic
    # underlying libproj not recent enough to support geod_geodesic
end
```